python-checks/src/main/java/org/sonar/python/checks/NumpyBusDayHaveValidMaskCheck.java

```diff
@@ -0,0 +1,131 @@
+import org.sonar.plugins.python.api.tree.StringLiteral;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.checks.utils.Expressions;
+import org.sonar.python.tree.TreeUtils;
+
+@Rule(key = "S6900")
+public class NumpyBusDayHaveValidMaskCheck extends PythonSubscriptionCheck {
+  private static final Pattern PATTERN_STRING1 = Pattern.compile("^[01]{7}$");
+  private static final Pattern PATTERN_STRING2 = Pattern
+    .compile("^(Mon|Tue|Wed|Thu|Fri|Sat|Sun|\\s|\\\\t|\\\\n|\\\\x0b|\\\\x0c|\\\\r)*+$");
+  private static final String MESSAGE_ARRAY = "Array must have 7 elements, all of which are 0 or 1.";
+  private static final String MESSAGE_STRING = "String must be either 7 characters long and contain only 0 and 1, or contain abbreviated weekdays.";
+  private static final String MESSAGE_SECONDARY_LOCATION = "Invalid mask is created here.";
+  private static final Map<String, Integer> FUNCTIONS_PARAMETER_POSITION = Map.of(
+    "numpy.busday_offset", 3,
+    "numpy.busday_count", 2,
+    "numpy.is_busday", 1,
+    "numpy.busdaycalendar", 0);
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, NumpyBusDayHaveValidMaskCheck::checkCallExpr);
+  }
+
+  private static void checkCallExpr(SubscriptionContext context) {
+    CallExpression callExpression = (CallExpression) context.syntaxNode();
+    Symbol calleeSymbol = callExpression.calleeSymbol();
+    if (calleeSymbol == null) {
+      return;
+    }
+    String fullyQualifiedName = calleeSymbol.fullyQualifiedName();
+    if (fullyQualifiedName == null) {
+      return;
+    }
+    Integer parameterPosition = FUNCTIONS_PARAMETER_POSITION.get(fullyQualifiedName);
+    if (parameterPosition == null) {
+      return;
+    }
+
+    RegularArgument weekmaskArgument = TreeUtils.nthArgumentOrKeyword(parameterPosition, "weekmask", callExpression.arguments());
+    if (weekmaskArgument == null) {
+      return;
+    }
+
+    if (weekmaskArgument.expression().is(Tree.Kind.STRING_LITERAL)) {
+      checkString(context, ((StringLiteral) weekmaskArgument.expression()).trimmedQuotesValue(), weekmaskArgument.expression());
+    } else if (weekmaskArgument.expression().is(Tree.Kind.LIST_LITERAL)) {
+      checkList(context, ((ListLiteral) weekmaskArgument.expression()));
+    } else if (weekmaskArgument.expression().is(Tree.Kind.NAME)) {
+      Name name = (Name) weekmaskArgument.expression();
+      Expressions.singleAssignedNonNameValue(name).ifPresent(assignedExpression -> {
+        if (assignedExpression.is(Tree.Kind.STRING_LITERAL)) {
+          checkString(context, ((StringLiteral) assignedExpression).trimmedQuotesValue(), weekmaskArgument.expression(), assignedExpression);
+        } else if (assignedExpression.is(Tree.Kind.LIST_LITERAL)) {
+          checkList(context, ((ListLiteral) assignedExpression), weekmaskArgument.expression(), assignedExpression);
+        }
+      });
```

Comment:
Could simply be done like this:
```java
    CallExpression callExpression = (CallExpression) context.syntaxNode();

    Optional.ofNullable(callExpression.calleeSymbol())
      .map(Symbol::fullyQualifiedName)
      .filter(FUNCTIONS_PARAMETER_POSITION::containsKey)
      .map(FUNCTIONS_PARAMETER_POSITION::get)
      .map(position -> TreeUtils.nthArgumentOrKeyword(position, "weekmask", callExpression.arguments()))
      .map(RegularArgument::expression)
      .ifPresent(weekmaskArgumentExpression -> {
        var value = Expressions.ifNameGetSingleAssignedNonNameValue(weekmaskArgumentExpression);
        value.flatMap(TreeUtils.toOptionalInstanceOfMapper(StringLiteral.class))
          .ifPresent(string -> checkString(context, string.trimmedQuotesValue(), weekmaskArgumentExpression, string));
        value.flatMap(TreeUtils.toOptionalInstanceOfMapper(ListLiteral.class))
          .ifPresent(array -> checkList(context, array, weekmaskArgumentExpression, array));
      });
```
and change the method `createIssue` to:
```java
 private static void createIssue(SubscriptionContext context, String message, Tree primaryLocation, @Nullable Tree secondaryLocation) {
    PreciseIssue issue = context.addIssue(primaryLocation, message);
    if (secondaryLocation != null && secondaryLocation != primaryLocation) {
      issue.secondary(secondaryLocation, MESSAGE_SECONDARY_LOCATION);
    }
  }
```
and then you don't need methods:
- `private static void checkString(SubscriptionContext context, String string, Tree primaryLocation)`
- `private static void checkList(SubscriptionContext context, ListLiteral listLiteral)`